inc/saiswitch.h

```diff
@@ -310,6 +310,9 @@ typedef enum _sai_switch_type_t
     /** Switch type is Fabric switch device */
     SAI_SWITCH_TYPE_FABRIC,
 
+    /** Switch type is POE (Power over Ethernet) */
+    SAI_SWITCH_TYPE_POE,
```

Comment:
i think this we would need to discuss, this on SAI community meeting, i think this is fine, but i think this is more like ability, since for example maybe NPU switch could have some ethernet ports that supports PoE, not sure if any exists, but i think we could discuss this, i think PoE is more like a "feature" of the switch, rather then entire switch on its own, but maybe i could be wrong